### PR TITLE
Bundler

### PR DESCRIPTION
**Pull request by delaneyj**
image

Add a bundler built into the website and allows user to pick and choose the plugins they need

### CODE REVIEW

**Review by zangster300 — COMMENTED** [review]
1.3k+ additions and only caught one typo, that's pretty impressive. New plugins restructure is phenomenal. LGTM

File: code/ts/library/src/plugins/preprocessors/core/actions.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import { PreprocessorPlugin, RegexpGroups } from "library/src/engine";
-import { wholePrefixSuffix } from "library/src/utils/regex";
+import { PreprocessorPlugin, RegexpGroups } from "../../../engine";
+import { wholePrefixSuffix } from "../../../utils/regex";
 
 // Replacing $$action(args) with ctx.actions.action(ctx, args)
 export const ActionsProcessorPlugin: PreprocessorPlugin = {
-    name: "action",
+    name: "👾action",
```
Comment (bencroker) [inline code/ts/library/src/plugins/preprocessors/core/actions.ts:6]: What is this little fellow?

> name: "👾action",

**Comment by bencroker** [comment]
This is looking great! Some questions...

1. Can you merge `main` into the PR? I noticed a few missing plugins.
2. Does the standard bundle include all plugins?
3. Should we perhaps build for a version of ECMAScript prior to ES2023 for better compatibility?

File: code/ts/library/src/plugins/preprocessors/core/actions.ts (hunk 1 above)
Comment (delaneyj) [reply code/ts/library/src/plugins/preprocessors/core/actions.ts]: Me trying to use emojis and seeing what the issues were... removed in latest

File: code/go/site/routes_bundler.go [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+
+var (
+	datastarBundlerRegexp = regexp.MustCompile(`import { (?P<name>[^"]*) } from "(?P<path>[^"]*)";`)
+	pluginTypeRegexp      = regexp.MustCompile(`pluginType: "(?P<name>.*)",`)
+	nameRegexp            = regexp.MustCompile(`name: "(?P<name>.*)",`)
+	prefixRegexp          = regexp.MustCompile(`prefix: "(?P<name>.*)",`)
+)
+
+type BundlerStore struct {
+	IncludedPlugines map[string]bool `json:"includedPlugins"`
```
[the remark this answers is not in the document]
Comment (delaneyj) [reply code/go/site/routes_bundler.go:35]: fixed in latest

**Comment by bencroker** [comment]
As per your Discord comment, ES2023 is supported everywhere 🚀

> https://caniuse.com/?feats=mdn-javascript_builtins_array_findlast,mdn-javascript_builtins_array_findlastindex,mdn-javascript_grammar_hashbang_comments,mdn-javascript_builtins_weakmap_symbol_as_keys,mdn-javascript_builtins_array_toreversed,mdn-javascript_builtins_array_tosorted,mdn-javascript_builtins_array_tospliced,mdn-javascript_builtins_array_with

**Pull request merged**